framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -113,14 +111,14 @@
     public void adaptMethodType(AnnotatedTypeFactory typeFactory, ExpressionTree invocation, AnnotatedExecutableType methodType) {
         //do nothing
     }
-
+    // TODO: THIS IS A BIG VIOLATION OF Single Responsibility and SHOULD BE FIXED, IT IS SOLELY HERE
```

Comment:
It really applies to adaptMethodType and getArgumentTypes.   TypeArgumentInference should only have the one method infer.  Unfortunately, the structure of methodFromUse/findTypeArguments etc... does not make it easy to adapt the arguments.  The single responsibility InferTypeArguments should have is to infer type arguments, it should NOT be adapting the arguments before this.  If I recall correctly, getArgumentTypes should just be replaced with it's method body and removed.  I'd have to take a closer look at SPARTA before actually removing this.